src/Feersum/Bind.fs

```diff
@@ -43,7 +45,34 @@ with
         | SyntaxConstant.Character c -> BoundLiteral.Character c
         | SyntaxConstant.Boolean b -> BoundLiteral.Boolean b
         | SyntaxConstant.Str s -> BoundLiteral.Str s
-    
+
+/// The rename of a single element exported or imported from a library
+type BoundRename =
+    | Rename of string * string
+
+/// A single export from a library declration
+type BoundExportSet =
+    | Plain of string
+    | Renamed of BoundRename
+    | Error
+
+/// A single import from a library declration
+type BoundImportSet =
+    | Plain of string list
+    | Only of BoundImportSet * string list
+    | Except of BoundImportSet * string list
+    | Prefix of BoundImportSet * string
+    | Renamed of BoundImportSet * BoundRename list
+    | Error
+
+/// Library Declarations
+/// 
+/// A library is made up of a collection of library declarations
+type BoundLibraryDeclaration =
+    | Export of BoundExportSet list
+    | Import of BoundImportSet list
+    | Error
```

Comment:
These declarations aren't really 'bound' types. This should probably be moved into a separate `Libraries.fs`
and follow the pattern of `Macro`. We'd have a `Library.parse` that returns some more structured
representation. Binding the library would properly deal with scopes and actually bind the body.